src/containers/collaborations/sagas/addDataSetsToCollaboration.js

```diff
@@ -55,6 +64,14 @@ function* addDataSetsToCollaborationWorker(action :SequenceAction) :Saga<*> {
     );
     if (responseError) throw responseError;
 
+    if (dataSetIdsToLoad.length) {
```

Comment:
No change required. I feel like this will always be true ... you are adding every single id from `dataSetIdsToLoad`.